### DISCUSSION TITLE
DomPDF not properly response when the data will more the 2000 thousand

### DISCUSSION THREAD

**Discussion by bimahlpatel** [dompdf/dompdf#2709] [General]
I have accounting data arround 10000 and I will create a PDF using dompdf in codeigniter framework, so, please any one help to how to create pdf easylyt to more the data

**Comment by bsweeney** [comment]
Given a large enough execution time and memory limit Dompdf could probably generate a PDF with that many records. Barring that, I've found the best way to do this right now is to generate the PDF in chunks and then combine the resulting PDFs at the end of the process.

**Comment by markkimsal** [comment]
You can try to use your RAM more efficiently with "unbuffered" queries

```php
$db->resultMode = MYSQLI_USE_RESULT; // for unbuffered results

$query = $db->query("YOUR QUERY");

$file = new \CodeIgniter\Files\File(WRITEPATH.'data.csv');
$csv = $file->openFile('w');

while ($row = $query->getUnbufferedRow('array')) {
    $csv->fputcsv($row);
}

$db->resultMode = MYSQLI_STORE_RESULT; // return to default mode
```
